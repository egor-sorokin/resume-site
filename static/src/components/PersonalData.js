import React, { Component } from 'react';
import withFetching from '../utils/api';
import { URL_PATH_PERSONAL_DATA } from '../constants/index';


const PersonalData = ({ data, isFetching, error}) => {
  const personData = data.personData || {};

  if (error) {
    return <p>{error.message}</p>;
  }

  if (isFetching) {
    return <p>Loading ...</p>;
  }

  return (
    <div>
      <section>
        <p>
          {personData.id}
        </p>
        <p>
          {personData.first_name}
        </p>
        <p>
          {personData.last_name}
        </p>
        <p>
          {personData.email}
        </p>
        <p>
          {personData.summary}
        </p>
      </section>
    </div>
  );
};

export default withFetching(URL_PATH_PERSONAL_DATA)(PersonalData);
