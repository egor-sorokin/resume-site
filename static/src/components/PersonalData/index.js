import React from 'react';
import PropTypes from 'prop-types';
import withFetching from '../../utils/api';
import { URL_PATH_PERSONAL_DATA } from '../../constants/index';


const PersonalData = ({ data, isFetching, error }) => {
  const personData = data.personData || {};

  if (error) {
    return <div><p>{error}</p></div>;
  }

  if (isFetching) {
    return <div><p>Loading...</p></div>;
  }

  return (
    <div>
      <section>
        <p>
          {personData.id}
        </p>
        <p>
          {personData.first_name}
        </p>
        <p>
          {personData.last_name}
        </p>
        <p>
          {personData.email}
        </p>
        <p>
          {personData.summary}
        </p>
      </section>
    </div>
  );
};

PersonalData.propTypes = {
  data: PropTypes.shape({
    id: PropTypes.number,
    first_name: PropTypes.string,
    last_name: PropTypes.string,
    email: PropTypes.string,
    summary: PropTypes.string,
  }),
  isFetching: PropTypes.bool,
  error: PropTypes.string,
};

PersonalData.defaultProps = {
  data: {},
  error: '',
  isFetching: false,
};


export default withFetching(URL_PATH_PERSONAL_DATA, PersonalData);
