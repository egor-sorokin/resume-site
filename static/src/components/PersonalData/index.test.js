import React from 'react';
import PersonalData from './index';
import { shallow, render, mount } from 'enzyme';

describe('<PersonalData />', () => {
  const component = shallow(<PersonalData />);

  it('renders without crashing', () => {
    shallow(<PersonalData />);
  });

  it('renders 1 <PersonalData /> component', () => {
    expect(component).toHaveLength(1);
  });

  it('matches snapshot', () => {
    expect(component).toMatchSnapshot();
  });

  it('renders children when passed in', () => {
    const wrapper = shallow((
      <PersonalData>
        <div className="unique" />
      </PersonalData>
    ));
    expect(wrapper.contains(<div className="unique" />)).toEqual(true);
  });

  it('calls componentDidMount', () => {
    const spy = jest.spyOn(PersonalData.prototype, 'componentDidMount');
    const wrapper = mount(<PersonalData />);
    expect(spy).toHaveBeenCalledTimes(1);

    afterEach(() => {
      spy.mockClear()
    });
  });
});