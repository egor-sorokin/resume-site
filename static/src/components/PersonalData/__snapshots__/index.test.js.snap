// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PersonalData /> matches snapshot 1`] = `
<PersonalData
  data={Object {}}
  error={null}
  isFetching={true}
/>
`;
